src/Components/ViewDialogModal/ViewDialogModal.tsx

```diff
@@ -0,0 +1,188 @@
+/** @format */
+
+import * as React from "react";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import ListItemText from "@mui/material/ListItemText";
+import ListItem from "@mui/material/ListItem";
+import List from "@mui/material/List";
+import Divider from "@mui/material/Divider";
+import AppBar from "@mui/material/AppBar";
+import Toolbar from "@mui/material/Toolbar";
+import IconButton from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import CloseIcon from "@mui/icons-material/Close";
+import Slide from "@mui/material/Slide";
+import {
+  Backdrop,
+  Box,
+  CircularProgress,
+  Grid,
+  ImageListItem,
+} from "@mui/material";
+import { defaultImage, imgView2, imgView3 } from "@/Assets/Images";
+import { useDispatch, useSelector } from "react-redux";
+import { viewRoomDetails } from "@/Redux/Features/ViewDetails/viewDetailsSlice";
+import { toast } from "react-toastify";
+// import axios from "axios";
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+const ViewDialogModal = ({ handleClose, open, itemId }) => {
+  const [currentImageIndex, setCurrentImageIndex] = React.useState(0);
+  const [roomDetailsData, setRoomDetailsData] = React.useState([]);
+  const [loading, setLoading] = React.useState(true);
+
+  const dispatch = useDispatch();
+  const images = [defaultImage, imgView2, imgView3];
+  const handleImageClick = () => {
+    setCurrentImageIndex((prevIndex) => (prevIndex + 1) % images.length);
+  };
+
+  // const viewItem = React.useCallback(async () => {
+  //   const viewDetails = await dispatch(viewRoomDetails(itemId));
+  //   setRoomDetailsData(viewDetails.payload.data.room);
+  // }, [dispatch, itemId]);
+  const viewItem = React.useCallback(async () => {

```

Comment:
You can remove the useCallback and add the function directly in the useEffect since we are only using it in the effect.